<!doctype html>
<html class="no-js" lang="en" data-content_root="../">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />

        <script async src="https://www.googletagmanager.com/gtag/js?id="></script>
        <script>
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            
            gtag('config', '');
            
        </script>
    <meta property="og:title" content="Dataset" />
<meta property="og:type" content="website" />
<meta property="og:url" content="configs/dataset.html" />
<meta property="og:site_name" content="Protein Workshop" />
<meta property="og:description" content="Summary of dataset sources To switch between different datasets, simply change the dataset argument in the launch command. For example: Where<DATASET_NAME> is given by bracketed name in the listing..." />
<meta property="og:image:width" content="1146" />
<meta property="og:image:height" content="600" />
<meta property="og:image" content="/_images/social_previews/summary_configs_dataset_8d050e58.png" />
<meta property="og:image:alt" content="Summary of dataset sources To switch between different datasets, simply change the dataset argument in the launch command. For example: Where<DATASET_NAME> i..." />
<meta name="description" content="Summary of dataset sources To switch between different datasets, simply change the dataset argument in the launch command. For example: Where<DATASET_NAME> is given by bracketed name in the listing..." />
<meta name="twitter:card" content="summary_large_image" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="Models" href="model.html" /><link rel="prev" title="Config Templates" href="templates.html" />

    <!-- Generated with Sphinx 7.2.4 and Furo 2023.08.19 -->
        <title>Dataset - Protein Workshop 0.2.0</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=135e06be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/tabs.css?v=4c969af8" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-codeautolink.css?v=125d5c1c" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=36a5483c" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">Protein Workshop 0.2.0</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/workshop_icon.png" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">Protein Workshop 0.2.0</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Architectural Overview</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../quickstart.html">Quickstart</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of Quickstart</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../quickstart_component/pretrain.html">Pre-Training</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quickstart_component/downstream.html">Downstream</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quickstart_component/download.html">Downloading Datasets</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Default Configs</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="templates.html">Config Templates</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="model.html">Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="task.html">Tasks</a></li>
<li class="toctree-l1"><a class="reference internal" href="features.html">Features</a></li>
<li class="toctree-l1"><a class="reference internal" href="transforms.html">Transforms</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../framework.html">Framework Components</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle navigation of Framework Components</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="framework_components/callbacks.html">Callbacks</a></li>
<li class="toctree-l2"><a class="reference internal" href="framework_components/debug.html">Debug</a></li>
<li class="toctree-l2"><a class="reference internal" href="framework_components/env.html">Environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="framework_components/logger.html">Loggers</a></li>
<li class="toctree-l2"><a class="reference internal" href="framework_components/hydra.html">Hydra Configuration</a></li>
<li class="toctree-l2"><a class="reference internal" href="framework_components/extras.html">Extras</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../ml_components.html">ML Components</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle navigation of ML Components</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="ml_components/trainer.html">Trainer</a></li>
<li class="toctree-l2"><a class="reference internal" href="ml_components/optimiser.html">Optimiser</a></li>
<li class="toctree-l2"><a class="reference internal" href="ml_components/scheduler.html">Schedulers</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.train.html">protein_workshop.train</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.datasets.html">protein_workshop.datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.models.html">protein_workshop.models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.tasks.html">protein_workshop.tasks</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.features.html">protein_workshop.features</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.utils.html">protein_workshop.utils</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/proteinworkshop.types.html">protein_workshop.types</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="dataset">
<h1>Dataset<a class="headerlink" href="#dataset" title="Link to this heading">#</a></h1>
<a class="reference internal image-reference" href="../_images/box_datasets.png"><img alt="Summary of dataset sources" class="align-center" src="../_images/box_datasets.png" style="width: 400px;" /></a>
<p>To switch between different datasets, simply change the <cite>dataset</cite> argument in the launch command. For example:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>workshop<span class="w"> </span>train<span class="w"> </span><span class="nv">encoder</span><span class="o">=</span>gear_net<span class="w"> </span><span class="nv">dataset</span><span class="o">=</span>&lt;DATASET_NAME&gt;<span class="w"> </span><span class="nv">task</span><span class="o">=</span>inverse_folding<span class="w"> </span><span class="nv">trainer</span><span class="o">=</span>cpu
<span class="c1"># or</span>
python<span class="w"> </span>proteinworkshop/train.py<span class="w"> </span><span class="nv">encoder</span><span class="o">=</span>gear_net<span class="w"> </span><span class="nv">dataset</span><span class="o">=</span>&lt;DATASET_NAME&gt;<span class="w"> </span><span class="nv">task</span><span class="o">=</span>inverse_folding<span class="w"> </span><span class="nv">trainer</span><span class="o">=</span>cpu<span class="w"> </span><span class="c1"># or trainer=gpu</span>
</pre></div>
</div>
<p>Where <code class="docutils literal notranslate"><span class="pre">&lt;DATASET_NAME&gt;</span></code> is given by bracketed name in the listing below. For example, the dataset name for CATH is <code class="docutils literal notranslate"><span class="pre">cath</span></code>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you have pip-installed <cite>proteinworkshop</cite>, you can download pre-training or processed downstream datasets from Zenodo with:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>workshop<span class="w"> </span>download<span class="w"> </span>&lt;DATASET_NAME&gt;
</pre></div>
</div>
</div>
<section id="unlabelled-datasets">
<h2>Unlabelled Datasets<a class="headerlink" href="#unlabelled-datasets" title="Link to this heading">#</a></h2>
<section id="structure-based-pre-training-corpuses">
<h3>Structure-based Pre-training Corpuses<a class="headerlink" href="#structure-based-pre-training-corpuses" title="Link to this heading">#</a></h3>
<p>Pre-training corpuses (with the exception of <code class="docutils literal notranslate"><span class="pre">pdb</span></code>, <code class="docutils literal notranslate"><span class="pre">cath</span></code>, and <code class="docutils literal notranslate"><span class="pre">astral</span></code>) are provided in FoldComp database format. This format is highly compressed, resulting in very small disk space requirements despite the large size. <code class="docutils literal notranslate"><span class="pre">pdb</span></code> is provided as a collection of
<code class="docutils literal notranslate"><span class="pre">MMTF</span></code> files, which are significantly smaller in size than conventional <code class="docutils literal notranslate"><span class="pre">.pdb</span></code> or <code class="docutils literal notranslate"><span class="pre">.cif</span></code> file.</p>
<div class="table-wrapper docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Source</p></th>
<th class="head"><p>Size</p></th>
<th class="head"><p>Disk Size</p></th>
<th class="head"><p>License</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">astral</span></code></p></td>
<td><p><a class="reference external" href="https://scop.berkeley.edu/">SCOPe</a> domain structures</p></td>
<td><p><a class="reference external" href="https://scop.berkeley.edu/">SCOPe/ASTRAL</a></p></td>
<td></td>
<td><p>1 - 2.2 Gb</p></td>
<td><p><a class="reference external" href="https://scop.berkeley.edu/about/ver=2.08">Publicly available</a></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">afdb_rep_v4</span></code></p></td>
<td><p>Representative structures identified from the <a class="reference external" href="https://alphafold.com/">AlphaFold database</a> by <a class="reference external" href="https://github.com/steineggerlab/foldseek">FoldSeek</a> structural clustering</p></td>
<td><p><a class="reference external" href="https://www.biorxiv.org/content/10.1101/2023.03.09.531927v1">Barrio-Hernandez et al.</a></p></td>
<td><p>2.27M Chains</p></td>
<td><p>9.6 Gb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">afdb_rep_dark_v4</span></code></p></td>
<td><p>Dark proteome structures identied by structural clustering of the <a class="reference external" href="https://alphafold.com/">AlphaFold database</a>.</p></td>
<td><p><a class="reference external" href="https://www.biorxiv.org/content/10.1101/2023.03.09.531927v1">Barrio-Hernandez et al.</a></p></td>
<td><p>~800k</p></td>
<td><p>2.2 Gb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">afdb_swissprot_v4</span></code></p></td>
<td><p>AlphaFold2 predictions for <a class="reference external" href="https://www.uniprot.org/help/uniprotkb">SwissProt/UniProtKB</a></p></td>
<td><p><a class="reference external" href="https://academic.oup.com/bioinformatics/article/39/4/btad153/7085592">Kim et al.</a></p></td>
<td><p>542k Chains</p></td>
<td><p>2.9 Gb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">afdb_uniprot_v4</span></code></p></td>
<td><p>AlphaFold2 predictions for <a class="reference external" href="https://www.uniprot.org/">UniProt</a></p></td>
<td><p><a class="reference external" href="https://academic.oup.com/bioinformatics/article/39/4/btad153/7085592">Kim et al.</a></p></td>
<td><p>214M Chains</p></td>
<td><p>1 Tb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a> / <a class="reference external" href="https://alphafold.ebi.ac.uk/assets/License-Disclaimer.pdf">CC-BY 4.0</a></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">cath</span></code></p></td>
<td><p><a class="reference external" href="https://www.cathdb.info/">CATH</a> 4.2 40% split by CATH topologies.</p></td>
<td><p><a class="reference external" href="https://www.mit.edu/~vgarg/GenerativeModelsForProteinDesign.pdf">Ingraham et al.</a></p></td>
<td><p>~18k chains</p></td>
<td><p>4.3 Gb</p></td>
<td><p><a class="reference external" href="http://cathdb.info/">CC-BY 4.0</a></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">esmatlas</span></code></p></td>
<td><p><a class="reference external" href="https://esmatlas.com/">ESMAtlas</a> predictions  (full)</p></td>
<td><p><a class="reference external" href="https://academic.oup.com/bioinformatics/article/39/4/btad153/7085592">Kim et al.</a></p></td>
<td></td>
<td><p>1 Tb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a> / <a class="reference external" href="https://esmatlas.com/about">CC-BY 4.0</a></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">esmatlas_v2023_02</span></code></p></td>
<td><p><a class="reference external" href="https://esmatlas.com/">ESMAtlas</a> predictions (v2023_02 release)</p></td>
<td><p><a class="reference external" href="https://academic.oup.com/bioinformatics/article/39/4/btad153/7085592">Kim et al.</a></p></td>
<td></td>
<td><p>137 Gb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a> / <a class="reference external" href="https://esmatlas.com/about">CC-BY 4.0</a></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">highquality_clust30</span></code></p></td>
<td><p><a class="reference external" href="https://esmatlas.com/">ESMAtlas</a> High Quality predictions</p></td>
<td><p><a class="reference external" href="https://academic.oup.com/bioinformatics/article/39/4/btad153/7085592">Kim et al.</a></p></td>
<td><p>37M Chains</p></td>
<td><p>114 Gb</p></td>
<td><p><a class="reference external" href="https://github.com/steineggerlab/foldcomp/blob/master/LICENSE.txt">GPL-3.0</a> / <a class="reference external" href="https://esmatlas.com/about">CC-BY 4.0</a></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">pdb</span></code></p></td>
<td><p>Experimental structures deposited in the <a class="reference external" href="https://www.rcsb.org/">RCSB Protein Data Bank</a></p></td>
<td><p><a class="reference external" href="https://academic.oup.com/nar/article/47/D1/D520/5144142">wwPDB consortium</a></p></td>
<td><p>~800k Chains</p></td>
<td><p>23 Gb</p></td>
<td><p><a class="reference external" href="https://www.rcsb.org/news/feature/611e8d97ef055f03d1f222c6">CC0 1.0</a></p></td>
</tr>
</tbody>
</table>
</div>
<details>
  <summary>Additionally, we provide several species-specific compilations (mostly reference species)</summary>

  | Name            | Description   | Source | Size |
  | ----------------| ----------- | ------ | ------ |
  | `a_thaliana`    | _Arabidopsis thaliana_ (thale cress) proteome | AlphaFold2|
  | `c_albicans`    | _Candida albicans_ (a fungus) proteome | AlphaFold2|
  | `c_elegans`     | _Caenorhabditis elegans_ (roundworm) proteome        | AlphaFold2       | |
  | `d_discoideum`  | _Dictyostelium discoideum_ (slime mold) proteome | AlphaFold2| |
  | `d_melanogaster`  | [_Drosophila melanogaster_](https://www.uniprot.org/taxonomy/7227) (fruit fly) proteome        | AlphaFold2        | |
  | `d_rerio`  | [_Danio rerio_](https://www.uniprot.org/taxonomy/7955) (zebrafish) proteome        | AlphaFold2        | |
  | `e_coli`  | _Escherichia coli_ (a bacteria) proteome        |  AlphaFold2       | |
  | `g_max`  | _Glycine max_ (soy bean) proteome       | AlphaFold2        | |
  | `h_sapiens`  | _Homo sapiens_ (human) proteome       |  AlphaFold2       | |
  | `m_jannaschii`  | _Methanocaldococcus jannaschii_ (an archaea) proteome        |  AlphaFold2       | |
  | `m_musculus`  | _Mus musculus_ (mouse) proteome        |   AlphaFold2      | |
  | `o_sativa`  | _Oryza sative_ (rice) proteome     |   AlphaFold2      | |
  | `r_norvegicus`  | _Rattus norvegicus_ (brown rat) proteome       |   AlphaFold2      | |
  | `s_cerevisiae`  | _Saccharomyces cerevisiae_ (brewer's yeast) proteome       |   AlphaFold2      | |
  | `s_pombe`  | _Schizosaccharomyces pombe_ (a fungus) proteome      |  AlphaFold2       | |
  | `z_mays`  | _Zea mays_ (corn) proteome    |   AlphaFold2      | |

</details></section>
<section id="astral-astral">
<h3><a class="reference internal" href="../modules/proteinworkshop.datasets.html#proteinworkshop.datasets.astral.AstralDataModule" title="proteinworkshop.datasets.astral.AstralDataModule"><code class="xref py py-class docutils literal notranslate"><span class="pre">ASTRAL</span></code></a> (<code class="docutils literal notranslate"><span class="pre">astral</span></code>)<a class="headerlink" href="#astral-astral" title="Link to this heading">#</a></h3>
<p>ASTRAL provides compendia of protein domain structures, regions of proteins that
can maintain their structure and function independently of the rest of the protein. Domains typically
exhibit highly-specific functions and can be considered structural building blocks of proteins.</p>
<div class="literal-block-wrapper docutils container" id="id21">
<div class="code-block-caption"><span class="caption-text">config/dataset/astral.yaml</span><a class="headerlink" href="#id21" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;proteinworkshop.datasets.astral.AstralDataModule&quot;</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/Astral/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">release</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;2.08&quot;</span><span class="w"> </span><span class="c1"># Version of ASTRAL to use</span>
<span class="w">  </span><span class="nt">identity</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;95&quot;</span><span class="w"> </span><span class="c1"># Percent identity clustering threshold to use</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">dataset_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of dataset to use</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="w">  </span><span class="nt">overwrite</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Whether to overwrite cached dataset example files</span>
<span class="w">  </span><span class="nt">train_val_test</span><span class="p">:</span><span class="w"> </span><span class="p p-Indicator">[</span><span class="nv">0.8</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="nv">0.1</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="nv">0.1</span><span class="p p-Indicator">]</span><span class="w"> </span><span class="c1"># Cross-validation ratios to use for train, val, and test splits</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">null</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="cath-cath">
<h3><a class="reference internal" href="../modules/proteinworkshop.datasets.html#proteinworkshop.datasets.cath.CATHDataModule" title="proteinworkshop.datasets.cath.CATHDataModule"><code class="xref py py-class docutils literal notranslate"><span class="pre">CATH</span></code></a> (<code class="docutils literal notranslate"><span class="pre">cath</span></code>)<a class="headerlink" href="#cath-cath" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id22">
<div class="code-block-caption"><span class="caption-text">config/dataset/cath.yaml</span><a class="headerlink" href="#id22" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;proteinworkshop.datasets.cath.CATHDataModule&quot;</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/cath/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Directory where raw PDB/mmtf files are stored</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the raw PDB/MMTF files</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">dataset_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">23</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="pdb-pdb">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">PDB</span></code> (<code class="docutils literal notranslate"><span class="pre">pdb</span></code>)<a class="headerlink" href="#pdb-pdb" title="Link to this heading">#</a></h3>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><code class="xref py py-class docutils literal notranslate"><span class="pre">proteinworkshop.datasets.pdb_dataset.PDBData</span></code></p>
</div>
<div class="literal-block-wrapper docutils container" id="id23">
<div class="code-block-caption"><span class="caption-text">config/dataset/pdb.yaml</span><a class="headerlink" href="#id23" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;proteinworkshop.datasets.pdb_dataset.PDBDataModule&quot;</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>

<span class="w">  </span><span class="nt">pdb_dataset</span><span class="p">:</span>
<span class="w">    </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;proteinworkshop.datasets.pdb_dataset.PDBData&quot;</span>
<span class="w">    </span><span class="nt">fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of dataset to use</span>
<span class="w">    </span><span class="nt">molecule_type</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;protein&quot;</span><span class="w"> </span><span class="c1"># Type of molecule for which to select</span>
<span class="w">    </span><span class="nt">experiment_types</span><span class="p">:</span><span class="w"> </span><span class="p p-Indicator">[</span><span class="s">&quot;diffraction&quot;</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="s">&quot;NMR&quot;</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="s">&quot;EM&quot;</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="s">&quot;other&quot;</span><span class="p p-Indicator">]</span><span class="w"> </span><span class="c1"># All experiment types</span>
<span class="w">    </span><span class="nt">max_length</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1000</span><span class="w"> </span><span class="c1"># Exclude polypeptides greater than length 1000</span>
<span class="w">    </span><span class="nt">min_length</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">10</span><span class="w"> </span><span class="c1"># Exclude peptides of length 10</span>
<span class="w">    </span><span class="nt">oligomeric_min</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1</span><span class="w"> </span><span class="c1"># Include only monomeric proteins</span>
<span class="w">    </span><span class="nt">oligomeric_max</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">5</span><span class="w"> </span><span class="c1"># Include up to 5-meric proteins </span>
<span class="w">    </span><span class="nt">best_resolution</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.0</span><span class="w"> </span><span class="c1"># Include only proteins with resolution &gt;= 0.0</span>
<span class="w">    </span><span class="nt">worst_resolution</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">8.0</span><span class="w"> </span><span class="c1"># Include only proteins with resolution &lt;= 8.0</span>
<span class="w">    </span><span class="nt">has_ligands</span><span class="p">:</span><span class="w"> </span><span class="p p-Indicator">[</span><span class="s">&quot;ZN&quot;</span><span class="p p-Indicator">]</span><span class="w"> </span><span class="c1"># Include only proteins containing the ligand `ZN`</span>
<span class="w">    </span><span class="nt">remove_ligands</span><span class="p">:</span><span class="w"> </span><span class="p p-Indicator">[]</span><span class="w"> </span><span class="c1"># Exclude specific ligands from any available protein-ligand complexes</span>
<span class="w">    </span><span class="nt">remove_non_standard_residues</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Include only proteins containing standard amino acid residues</span>
<span class="w">    </span><span class="nt">remove_pdb_unavailable</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Include only proteins that are available to download</span>
<span class="w">    </span><span class="nt">split_sizes</span><span class="p">:</span><span class="w"> </span><span class="p p-Indicator">[</span><span class="nv">0.8</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="nv">0.1</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="nv">0.1</span><span class="p p-Indicator">]</span><span class="w"> </span><span class="c1"># Cross-validation ratios to use for train, val, and test splits</span>
</pre></div>
</div>
</div>
</section>
<section id="afdb-rep-v4-afdb-rep-v4">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">AFdb</span> <span class="pre">Rep.</span> <span class="pre">v4</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_rep_v4</span></code>)<a class="headerlink" href="#afdb-rep-v4-afdb-rep-v4" title="Link to this heading">#</a></h3>
<p>This is a dataset of approximately 3 million protein structures from the AlphaFold database, structurally clustered using FoldSeek.</p>
<div class="literal-block-wrapper docutils container" id="id24">
<div class="code-block-caption"><span class="caption-text">config/dataset/afdb_rep_v4.yaml</span><a class="headerlink" href="#id24" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">graphein.ml.datasets.foldcomp_dataset.FoldCompLightningDataModule</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/afdb_rep_v4/</span>
<span class="w">  </span><span class="nt">database</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_rep_v4&quot;</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>

<span class="w">  </span><span class="nt">train_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.98</span>
<span class="w">  </span><span class="nt">val_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.01</span>
<span class="w">  </span><span class="nt">test_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.01</span>

<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">use_graphein</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">transform</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_rep_v4&quot;</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">None</span><span class="w"> </span><span class="c1"># number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="afdb-dark-proteome-afdb-rep-dark-v4">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">AFdb</span> <span class="pre">Dark</span> <span class="pre">Proteome</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_rep_dark_v4</span></code>)<a class="headerlink" href="#afdb-dark-proteome-afdb-rep-dark-v4" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id25">
<div class="code-block-caption"><span class="caption-text">config/dataset/afdb_rep_dark_v4.yaml</span><a class="headerlink" href="#id25" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">graphein.ml.datasets.foldcomp_dataset.FoldCompLightningDataModule</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/afdb_rep_dark_v4/</span>
<span class="w">  </span><span class="nt">database</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_rep_dark_v4&quot;</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>

<span class="w">  </span><span class="nt">train_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span>
<span class="w">  </span><span class="nt">val_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>
<span class="w">  </span><span class="nt">test_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>

<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">use_graphein</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">transform</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_rep_dark_v4&quot;</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">None</span><span class="w"> </span><span class="c1"># number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="esm-atlas-esmatlas-v2023-02">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">ESM</span> <span class="pre">Atlas</span></code> (<code class="docutils literal notranslate"><span class="pre">esmatlas_v2023_02</span></code>)<a class="headerlink" href="#esm-atlas-esmatlas-v2023-02" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id26">
<div class="code-block-caption"><span class="caption-text">config/dataset/esmatlas_v2023_02.yaml</span><a class="headerlink" href="#id26" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">graphein.ml.datasets.foldcomp_dataset.FoldCompLightningDataModule</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/esmatlas_v2023_02/</span>
<span class="w">  </span><span class="nt">database</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;esmatlas_v2023_02&quot;</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>

<span class="w">  </span><span class="nt">train_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span>
<span class="w">  </span><span class="nt">val_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>
<span class="w">  </span><span class="nt">test_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>

<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">use_graphein</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">transform</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;esmatlas_v2023_02&quot;</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">None</span><span class="w"> </span><span class="c1"># number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="esm-atlas-high-quality-highquality-clust30">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">ESM</span> <span class="pre">Atlas</span> <span class="pre">(High</span> <span class="pre">Quality)</span></code> (<code class="docutils literal notranslate"><span class="pre">highquality_clust30</span></code>)<a class="headerlink" href="#esm-atlas-high-quality-highquality-clust30" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id27">
<div class="code-block-caption"><span class="caption-text">config/dataset/highquality_clust30.yaml</span><a class="headerlink" href="#id27" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">graphein.ml.datasets.foldcomp_dataset.FoldCompLightningDataModule</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/highquality_clust30/</span>
<span class="w">  </span><span class="nt">database</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;highquality_clust30&quot;</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>

<span class="w">  </span><span class="nt">train_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span>
<span class="w">  </span><span class="nt">val_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>
<span class="w">  </span><span class="nt">test_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>

<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">use_graphein</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">transform</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;highquality_clust30&quot;</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">None</span><span class="w"> </span><span class="c1"># number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="uniprot-alphafold-afdb-uniprot-v4">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">UniProt</span> <span class="pre">(Alphafold)</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_uniprot_v4</span></code>)<a class="headerlink" href="#uniprot-alphafold-afdb-uniprot-v4" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id28">
<div class="code-block-caption"><span class="caption-text">config/dataset/afdb_uniprot_v4.yaml</span><a class="headerlink" href="#id28" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">graphein.ml.datasets.foldcomp_dataset.FoldCompLightningDataModule</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/afdb_uniprot_v4/</span>
<span class="w">  </span><span class="nt">database</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_uniprot_v4&quot;</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>

<span class="w">  </span><span class="nt">train_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span>
<span class="w">  </span><span class="nt">val_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>
<span class="w">  </span><span class="nt">test_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>

<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">use_graphein</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">transform</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_uniprot_v4&quot;</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">None</span><span class="w"> </span><span class="c1"># number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="swissprot-alphafold-afdb-swissprot-v4">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">SwissProt</span> <span class="pre">(Alphafold)</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_swissprot_v4</span></code>)<a class="headerlink" href="#swissprot-alphafold-afdb-swissprot-v4" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id29">
<div class="code-block-caption"><span class="caption-text">config/dataset/afdb_swissprot_v4.yaml</span><a class="headerlink" href="#id29" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">graphein.ml.datasets.foldcomp_dataset.FoldCompLightningDataModule</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/afdb_swissprot_v4/</span>
<span class="w">  </span><span class="nt">database</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_swissprot_v4&quot;</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span>

<span class="w">  </span><span class="nt">train_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span>
<span class="w">  </span><span class="nt">val_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>
<span class="w">  </span><span class="nt">test_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span>

<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">use_graphein</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span>
<span class="w">  </span><span class="nt">transform</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;afdb_swissprot_v4&quot;</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">None</span><span class="w"> </span><span class="c1"># number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="species-specific-datasets">
<h3>Species-Specific Datasets<a class="headerlink" href="#species-specific-datasets" title="Link to this heading">#</a></h3>
<p>TODO</p>
</section>
</section>
<section id="graph-level-datasets">
<h2>Graph-level Datasets<a class="headerlink" href="#graph-level-datasets" title="Link to this heading">#</a></h2>
<section id="antibody-developability-antibody-developability">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">Antibody</span> <span class="pre">Developability</span></code> (<code class="docutils literal notranslate"><span class="pre">antibody_developability</span></code>)<a class="headerlink" href="#antibody-developability-antibody-developability" title="Link to this heading">#</a></h3>
<p>Therapeutic antibodies must be optimised for favourable
physicochemical properties in addition to target binding affinity and specificity
to be viable development candidates. Consequently, this task frames prediction of antibody
developability as a binary graph classification task indicating whether a given antibody is developable</p>
<p>Dataset: We adopt the antibody developability dataset originally curated from SabDab by TDC.</p>
<p>Impact: From a benchmarking perspective, this task is important as it enables targeted performance
assessment of models on a specific (immunoglobulin) fold, providing insight into whether general-
purpose structure-based encoders can be applicable to fold-specific tasks.</p>
<div class="literal-block-wrapper docutils container" id="id30">
<div class="code-block-caption"><span class="caption-text">config/dataset/antibody_developability.yaml</span><a class="headerlink" href="#id30" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.antibody_developability.AntibodyDevelopabilityDataModule</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/AntibodyDevelopability</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Path to all downloaded PDB files</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">in_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Load the dataset in memory</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the structure files</span>
<span class="w">  </span><span class="nt">obsolete_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="w"> </span><span class="c1"># What to do with obsolete PDB entries</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="atom3d-mutation-stability-prediction-atom3d-msp">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">Atom3D</span> <span class="pre">Mutation</span> <span class="pre">Stability</span> <span class="pre">Prediction</span></code> (<code class="docutils literal notranslate"><span class="pre">atom3d_msp</span></code>)<a class="headerlink" href="#atom3d-mutation-stability-prediction-atom3d-msp" title="Link to this heading">#</a></h3>
<p>This task is defined in the <a class="reference external" href="https://www.atom3d.ai/msp.html">Atom3D benchmark</a>.</p>
<p>As per their documentation:</p>
<p><strong>Impact</strong>: Identifying mutations that stabilize a protein’s interactions is a key task in designing new proteins. Experimental techniques for probing these are labor intensive, motivating the development of efficient computational methods.</p>
<p><strong>Dataset description</strong>: We derive a novel dataset by collecting single-point mutations from the SKEMPI database (Jankauskaitė et al., 2019) and model each mutation into the structure to produce mutated structures.</p>
<p><strong>Task</strong>: We formulate this as a binary classification task where we predict whether the stability of the complex increases as a result of the mutation.</p>
<p><strong>Splitting criteria</strong>: We split protein complexes by sequence identity at 30%.</p>
<p><strong>Downloads</strong>: The full dataset, split data, and split indices are available for download via Zenodo (doi:10.5281/zenodo.4962515)</p>
<div class="literal-block-wrapper docutils container" id="id31">
<div class="code-block-caption"><span class="caption-text">config/dataset/atom3d_msp.yaml</span><a class="headerlink" href="#id31" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.atom3d_datamodule.ATOM3DDataModule</span>
<span class="w">  </span><span class="nt">task</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">MSP</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ATOM3D</span>
<span class="w">  </span><span class="nt">max_units</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0</span>
<span class="w">  </span><span class="nt">unit</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">edge</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">false</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span>
</pre></div>
</div>
</div>
</section>
<section id="atom3d-protein-structure-ranking-atom3d-psr">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">Atom3D</span> <span class="pre">Protein</span> <span class="pre">Structure</span> <span class="pre">Ranking</span></code> (<code class="docutils literal notranslate"><span class="pre">atom3d_psr</span></code>)<a class="headerlink" href="#atom3d-protein-structure-ranking-atom3d-psr" title="Link to this heading">#</a></h3>
<p>This task is defined in the <a class="reference external" href="https://www.atom3d.ai/psr.html">Atom3D benchmark</a>.</p>
<p>As per their documentation:</p>
<p><strong>Impact</strong>: Proteins are one of the primary workhorses of the cell, and knowing their structure is often critical to understanding (and engineering) their function.</p>
<p><strong>Dataset description</strong>: The Critical Assessment of Structure Prediction (CASP) (Kryshtafovych et al., 2019) is a blind international competition for predicting protein structure.</p>
<p><strong>Task</strong>: We formulate this as a regression task, where we predict the global distance test (GDT_TS) from the true structure for each of the predicted structures submitted in the last 18 years of CASP.</p>
<p><strong>Splitting criteria</strong>: We split structures temporally by competition year.</p>
<p><strong>Downloads</strong>: The full dataset, split data, and split indices are available for download via Zenodo (doi:10.5281/zenodo.4915648)</p>
<div class="literal-block-wrapper docutils container" id="id32">
<div class="code-block-caption"><span class="caption-text">config/dataset/atom3d_psr.yaml</span><a class="headerlink" href="#id32" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.atom3d_datamodule.ATOM3DDataModule</span>
<span class="w">  </span><span class="nt">task</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">PSR</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ATOM3D</span>
<span class="w">  </span><span class="nt">max_units</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0</span>
<span class="w">  </span><span class="nt">unit</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">edge</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">false</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1</span>
</pre></div>
</div>
</div>
</section>
<section id="deep-sea-protein-classification-deep-sea-proteins">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">Deep</span> <span class="pre">Sea</span> <span class="pre">Protein</span> <span class="pre">Classification</span></code> (<code class="docutils literal notranslate"><span class="pre">deep_sea_proteins</span></code>)<a class="headerlink" href="#deep-sea-protein-classification-deep-sea-proteins" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id33">
<div class="code-block-caption"><span class="caption-text">config/dataset/deepsea.yaml</span><a class="headerlink" href="#id33" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.deep_sea_proteins.DeepSeaProteinsDataModule</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/deep-sea-proteins/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Directory where raw PDB/mmtf files are stored</span>
<span class="w">  </span><span class="nt">validation_fold</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;4&quot;</span><span class="w"> </span><span class="c1"># Fold to use for validation (one of &#39;1&#39;, &#39;2&#39;, &#39;3&#39;, &#39;4&#39;, &#39;PM_group&#39;)</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">8</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">obsolete_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;drop&quot;</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the raw PDB/MMTF files</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span>
</pre></div>
</div>
</div>
</section>
<section id="enzyme-commission-number-prediction-ec-reaction">
<h3><code class="xref py py-class docutils literal notranslate"><span class="pre">Enzyme</span> <span class="pre">Commission</span> <span class="pre">Number</span> <span class="pre">Prediction</span></code> (<code class="docutils literal notranslate"><span class="pre">ec_reaction</span></code>)<a class="headerlink" href="#enzyme-commission-number-prediction-ec-reaction" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id34">
<div class="code-block-caption"><span class="caption-text">config/dataset/ec_reaction.yaml</span><a class="headerlink" href="#id34" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.ec_reaction.EnzymeCommissionReactionDataset</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ECReaction/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Directory where raw PDB/mmtf files are stored</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the raw PDB/MMTF files</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">8</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">dataset_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use</span>
<span class="w">  </span><span class="nt">shuffle_labels</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Whether to shuffle labels for permutation testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">384</span>
</pre></div>
</div>
</div>
</section>
<section id="fold-classification-fold-family-fold-superfamily-fold-fold">
<h3><a class="reference internal" href="../modules/proteinworkshop.datasets.html#proteinworkshop.datasets.fold_classification.FoldClassificationDataModule" title="proteinworkshop.datasets.fold_classification.FoldClassificationDataModule"><code class="xref py py-class docutils literal notranslate"><span class="pre">Fold</span> <span class="pre">Classification</span></code></a> (<code class="docutils literal notranslate"><span class="pre">fold-family</span></code>, <code class="docutils literal notranslate"><span class="pre">fold-superfamily</span></code>, <code class="docutils literal notranslate"><span class="pre">fold-fold</span></code>)<a class="headerlink" href="#fold-classification-fold-family-fold-superfamily-fold-fold" title="Link to this heading">#</a></h3>
<p>This is a multiclass graph classification task where each protein, G, is mapped to a
label y ∈ {1, … , 1195} denoting the fold class.</p>
<p><strong>Dataset</strong>: We adopt the fold classification dataset originally curated from SCOP 1.75 by Hermosilla et al. In
particular, this dataset contains three distinct test splits across which we average a method’s results.</p>
<p><strong>Impact</strong>: The utility of this task is that it serves as a litmus test for the ability of a model to distinguish
different structural folds. It stands to reason that models that perform poorly on distinguishing fold
classes likely learn limited or low-quality structural representations.214</p>
<p><strong>Splitting Criteria</strong>:</p>
<div class="literal-block-wrapper docutils container" id="id35">
<div class="code-block-caption"><span class="caption-text">config/dataset/fold_family.yaml</span><a class="headerlink" href="#id35" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;proteinworkshop.datasets.fold_classification.FoldClassificationDataModule&quot;</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/FoldClassification/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">split</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;family&quot;</span><span class="w"> </span><span class="c1"># Level of fold classification to perform (`family`, `superfamily`, or `fold`)</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">dataset_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of dataset to use</span>
<span class="w">  </span><span class="nt">shuffle_labels</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Whether to shuffle labels for permutation testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1195</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="gene-ontology-go-bp-go-cc-go-mf">
<h3>Gene Ontology (<code class="docutils literal notranslate"><span class="pre">go-bp</span></code>, <code class="docutils literal notranslate"><span class="pre">go-cc</span></code>, <code class="docutils literal notranslate"><span class="pre">go-mf</span></code>)<a class="headerlink" href="#gene-ontology-go-bp-go-cc-go-mf" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id36">
<div class="code-block-caption"><span class="caption-text">config/dataset/go-bp.yaml / config/dataset/go-cc.yaml / config/dataset/go-mf.yaml</span><a class="headerlink" href="#id36" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.go.GeneOntologyDataset</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/GeneOntology/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Directory where raw PDB/mmtf files are stored</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the raw PDB/MMTF files</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">dataset_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use</span>
<span class="w">  </span><span class="nt">shuffle_labels</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Whether to shuffle labels for permutation testing</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">8</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">split</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="w"> </span><span class="c1"># Split of the dataset to use (`BP`, `MF`, `CC`)</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
</section>
<section id="node-level-datasets">
<h2>Node-level Datasets<a class="headerlink" href="#node-level-datasets" title="Link to this heading">#</a></h2>
<section id="atom3d-residue-identity-prediction-atom3d-res">
<h3>Atom3D Residue Identity Prediction (<code class="docutils literal notranslate"><span class="pre">atom3d_res</span></code>)<a class="headerlink" href="#atom3d-residue-identity-prediction-atom3d-res" title="Link to this heading">#</a></h3>
<p>This task is defined in the <a class="reference external" href="https://www.atom3d.ai/res.html">Atom3D benchmark</a>.</p>
<p>As per their documentation:</p>
<p><strong>Impact</strong>: Understanding the structural role of individual amino acids is important for engineering new proteins. We can understand this role by predicting the substitutabilities of different amino acids at a given protein site based on the surrounding structural environment.</p>
<p><strong>Dataset description</strong>: We generate a novel dataset consisting of atomic environments extracted from nonredundant structures in the PDB.</p>
<p><strong>Task</strong>: We formulate this as a classification task where we predict the identity of the amino acid in the center of the environment based on all other atoms.</p>
<p><strong>Splitting criteria</strong>: We split residue environments by domain-level CATH protein topology class.</p>
<div class="literal-block-wrapper docutils container" id="id37">
<div class="code-block-caption"><span class="caption-text">config/dataset/atom3d_res.yaml</span><a class="headerlink" href="#id37" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.atom3d_datamodule.ATOM3DDataModule</span>
<span class="w">  </span><span class="nt">task</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">RES</span>
<span class="w">  </span><span class="nt">data_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ATOM3D</span>
<span class="w">  </span><span class="nt">res_split</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">cath-topology</span>
<span class="w">  </span><span class="nt">max_units</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0</span>
<span class="w">  </span><span class="nt">unit</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">edge</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">false</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">20</span>
</pre></div>
</div>
</div>
</section>
<section id="ccpdb-ligand-binding-ccpdb-ligand">
<h3>CCPDB Ligand Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_ligand</span></code>)<a class="headerlink" href="#ccpdb-ligand-binding-ccpdb-ligand" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id38">
<div class="code-block-caption"><span class="caption-text">config/dataset/ccpdb_ligand.yaml</span><a class="headerlink" href="#id38" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.cc_pdb.CCPDBDataModule</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ccpdb/ligands/</span><span class="w"> </span><span class="c1"># Path to the dataset</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Path to the PDB files</span>
<span class="w">  </span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;ligands&quot;</span><span class="w"> </span><span class="c1"># Name of the ccPDB dataset</span>

<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for the dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for the dataloader</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the structure files</span>
<span class="w">  </span><span class="nt">obsolete_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="w"> </span><span class="c1"># What to do with obsolete PDB entries</span>
<span class="w">  </span><span class="nt">split_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;random&quot;</span><span class="w"> </span><span class="c1"># (or &#39;stratified&#39;) How to split the dataset</span>
<span class="w">  </span><span class="nt">train_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for training</span>
<span class="w">  </span><span class="nt">val_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for validation</span>
<span class="w">  </span><span class="nt">test_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">7</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="ccpdb-metal-binding-ccpdb-metal">
<h3>CCPDB Metal Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_metal</span></code>)<a class="headerlink" href="#ccpdb-metal-binding-ccpdb-metal" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id39">
<div class="code-block-caption"><span class="caption-text">config/dataset/ccpdb_metal.yaml</span><a class="headerlink" href="#id39" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.cc_pdb.CCPDBDataModule</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ccpdb/metal/</span><span class="w"> </span><span class="c1"># Path to the dataset</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Path to the PDB files</span>
<span class="w">  </span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;metal&quot;</span><span class="w"> </span><span class="c1"># Name of the ccPDB dataset</span>

<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for the dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for the dataloader</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the structure files</span>
<span class="w">  </span><span class="nt">obsolete_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="w"> </span><span class="c1"># What to do with obsolete PDB entries</span>
<span class="w">  </span><span class="nt">split_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;random&quot;</span><span class="w"> </span><span class="c1"># (or &#39;stratified&#39;) How to split the dataset</span>
<span class="w">  </span><span class="nt">train_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for training</span>
<span class="w">  </span><span class="nt">val_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for validation</span>
<span class="w">  </span><span class="nt">test_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">7</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="ccpdb-nucleic-acid-binding-ccpdb-nucleic">
<h3>CCPDB Nucleic Acid Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_nucleic</span></code>)<a class="headerlink" href="#ccpdb-nucleic-acid-binding-ccpdb-nucleic" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id40">
<div class="code-block-caption"><span class="caption-text">config/dataset/ccpdb_nucleic.yaml</span><a class="headerlink" href="#id40" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.cc_pdb.CCPDBDataModule</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ccpdb/nucleic/</span><span class="w"> </span><span class="c1"># Path to the dataset</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Path to the PDB files</span>
<span class="w">  </span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;nucleic&quot;</span><span class="w"> </span><span class="c1"># Name of the ccPDB dataset</span>

<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for the dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for the dataloader</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the structure files</span>
<span class="w">  </span><span class="nt">obsolete_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="w"> </span><span class="c1"># What to do with obsolete PDB entries</span>
<span class="w">  </span><span class="nt">split_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;random&quot;</span><span class="w"> </span><span class="c1"># (or &#39;stratified&#39;) How to split the dataset</span>
<span class="w">  </span><span class="nt">train_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for training</span>
<span class="w">  </span><span class="nt">val_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for validation</span>
<span class="w">  </span><span class="nt">test_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="ccpdb-nucleotide-binding-ccpdb-nucleotides">
<h3>CCPDB Nucleotide Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_nucleotides</span></code>)<a class="headerlink" href="#ccpdb-nucleotide-binding-ccpdb-nucleotides" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id41">
<div class="code-block-caption"><span class="caption-text">config/dataset/ccpdb_nucleotides.yaml</span><a class="headerlink" href="#id41" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.cc_pdb.CCPDBDataModule</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/ccpdb/nucleotides/</span><span class="w"> </span><span class="c1"># Path to the dataset</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Path to the PDB files</span>
<span class="w">  </span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;nucleotides&quot;</span><span class="w"> </span><span class="c1"># Name of the ccPDB dataset</span>

<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for the dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4</span><span class="w"> </span><span class="c1"># Number of workers for the dataloader</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the structure files</span>
<span class="w">  </span><span class="nt">obsolete_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="w"> </span><span class="c1"># What to do with obsolete PDB entries</span>
<span class="w">  </span><span class="nt">split_strategy</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;random&quot;</span><span class="w"> </span><span class="c1"># (or &#39;stratified&#39;) How to split the dataset</span>
<span class="w">  </span><span class="nt">train_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.8</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for training</span>
<span class="w">  </span><span class="nt">val_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for validation</span>
<span class="w">  </span><span class="nt">test_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.1</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use for testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span>

<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">8</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="post-translational-modifications-ptm">
<h3>Post Translational Modifications (<code class="docutils literal notranslate"><span class="pre">ptm</span></code>)<a class="headerlink" href="#post-translational-modifications-ptm" title="Link to this heading">#</a></h3>
<div class="literal-block-wrapper docutils container" id="id42">
<div class="code-block-caption"><span class="caption-text">config/dataset/ptm.yaml</span><a class="headerlink" href="#id42" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;proteinworkshop.datasets.ptm.PTMDataModule&quot;</span>
<span class="w">  </span><span class="nt">dataset_name</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;ptm_13&quot;</span><span class="w"> </span><span class="c1"># Options currently include (`ptm_13`, `optm`)</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/PostTranslationalModification/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">in_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Load the dataset in memory</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">16</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">13</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
<section id="ppi-site-prediction-masif-site">
<h3>PPI Site Prediction (<code class="docutils literal notranslate"><span class="pre">masif_site</span></code>)<a class="headerlink" href="#ppi-site-prediction-masif-site" title="Link to this heading">#</a></h3>
<p>We use the dataset of experimental structures curated from the PDB by Gainza et al.
and retain the original splits, though we modify the labelling scheme to be based on inter-atomic
proximity (3.5 Å), which can be user-defined, rather than solvent exclusion.</p>
<p>The dataset is composed by selecting PPI pairs from the PRISM list of nonredundant proteins,
the ZDock benchmark, PDBBind and SabDab. Splits are performed using CD-HIT and structural splits
are performed using TM-algin.</p>
<div class="literal-block-wrapper docutils container" id="id43">
<div class="code-block-caption"><span class="caption-text">config/dataset/masif_site.yaml</span><a class="headerlink" href="#id43" title="Link to this code">#</a></div>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">datamodule</span><span class="p">:</span>
<span class="w">  </span><span class="nt">_target_</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">proteinworkshop.datasets.masif_site.MaSIFPPISP</span>
<span class="w">  </span><span class="nt">path</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/masif_site/</span><span class="w"> </span><span class="c1"># Directory where the dataset is stored</span>
<span class="w">  </span><span class="nt">pdb_dir</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${env.paths.data}/pdb/</span><span class="w"> </span><span class="c1"># Directory where raw PDB/mmtf files are stored</span>
<span class="w">  </span><span class="nt">format</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;mmtf&quot;</span><span class="w"> </span><span class="c1"># Format of the raw PDB/MMTF files</span>
<span class="w">  </span><span class="nt">batch_size</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">32</span><span class="w"> </span><span class="c1"># Batch size for dataloader</span>
<span class="w">  </span><span class="nt">pin_memory</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">True</span><span class="w"> </span><span class="c1"># Pin memory for dataloader</span>
<span class="w">  </span><span class="nt">num_workers</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">8</span><span class="w"> </span><span class="c1"># Number of workers for dataloader</span>
<span class="w">  </span><span class="nt">dataset_fraction</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.0</span><span class="w"> </span><span class="c1"># Fraction of the dataset to use</span>
<span class="w">  </span><span class="nt">shuffle_labels</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">False</span><span class="w"> </span><span class="c1"># Whether to shuffle labels for permutation testing</span>
<span class="w">  </span><span class="nt">transforms</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">${transforms}</span><span class="w"> </span><span class="c1"># Transforms to apply to dataset examples</span>
<span class="nt">num_classes</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span><span class="w"> </span><span class="c1"># Number of classes</span>
</pre></div>
</div>
</div>
</section>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="model.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Models</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="templates.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Config Templates</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2023, Arian R. Jamasb
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Dataset</a><ul>
<li><a class="reference internal" href="#unlabelled-datasets">Unlabelled Datasets</a><ul>
<li><a class="reference internal" href="#structure-based-pre-training-corpuses">Structure-based Pre-training Corpuses</a></li>
<li><a class="reference internal" href="#astral-astral"><code class="xref py py-class docutils literal notranslate"><span class="pre">ASTRAL</span></code> (<code class="docutils literal notranslate"><span class="pre">astral</span></code>)</a></li>
<li><a class="reference internal" href="#cath-cath"><code class="xref py py-class docutils literal notranslate"><span class="pre">CATH</span></code> (<code class="docutils literal notranslate"><span class="pre">cath</span></code>)</a></li>
<li><a class="reference internal" href="#pdb-pdb"><code class="xref py py-class docutils literal notranslate"><span class="pre">PDB</span></code> (<code class="docutils literal notranslate"><span class="pre">pdb</span></code>)</a></li>
<li><a class="reference internal" href="#afdb-rep-v4-afdb-rep-v4"><code class="xref py py-class docutils literal notranslate"><span class="pre">AFdb</span> <span class="pre">Rep.</span> <span class="pre">v4</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_rep_v4</span></code>)</a></li>
<li><a class="reference internal" href="#afdb-dark-proteome-afdb-rep-dark-v4"><code class="xref py py-class docutils literal notranslate"><span class="pre">AFdb</span> <span class="pre">Dark</span> <span class="pre">Proteome</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_rep_dark_v4</span></code>)</a></li>
<li><a class="reference internal" href="#esm-atlas-esmatlas-v2023-02"><code class="xref py py-class docutils literal notranslate"><span class="pre">ESM</span> <span class="pre">Atlas</span></code> (<code class="docutils literal notranslate"><span class="pre">esmatlas_v2023_02</span></code>)</a></li>
<li><a class="reference internal" href="#esm-atlas-high-quality-highquality-clust30"><code class="xref py py-class docutils literal notranslate"><span class="pre">ESM</span> <span class="pre">Atlas</span> <span class="pre">(High</span> <span class="pre">Quality)</span></code> (<code class="docutils literal notranslate"><span class="pre">highquality_clust30</span></code>)</a></li>
<li><a class="reference internal" href="#uniprot-alphafold-afdb-uniprot-v4"><code class="xref py py-class docutils literal notranslate"><span class="pre">UniProt</span> <span class="pre">(Alphafold)</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_uniprot_v4</span></code>)</a></li>
<li><a class="reference internal" href="#swissprot-alphafold-afdb-swissprot-v4"><code class="xref py py-class docutils literal notranslate"><span class="pre">SwissProt</span> <span class="pre">(Alphafold)</span></code> (<code class="docutils literal notranslate"><span class="pre">afdb_swissprot_v4</span></code>)</a></li>
<li><a class="reference internal" href="#species-specific-datasets">Species-Specific Datasets</a></li>
</ul>
</li>
<li><a class="reference internal" href="#graph-level-datasets">Graph-level Datasets</a><ul>
<li><a class="reference internal" href="#antibody-developability-antibody-developability"><code class="xref py py-class docutils literal notranslate"><span class="pre">Antibody</span> <span class="pre">Developability</span></code> (<code class="docutils literal notranslate"><span class="pre">antibody_developability</span></code>)</a></li>
<li><a class="reference internal" href="#atom3d-mutation-stability-prediction-atom3d-msp"><code class="xref py py-class docutils literal notranslate"><span class="pre">Atom3D</span> <span class="pre">Mutation</span> <span class="pre">Stability</span> <span class="pre">Prediction</span></code> (<code class="docutils literal notranslate"><span class="pre">atom3d_msp</span></code>)</a></li>
<li><a class="reference internal" href="#atom3d-protein-structure-ranking-atom3d-psr"><code class="xref py py-class docutils literal notranslate"><span class="pre">Atom3D</span> <span class="pre">Protein</span> <span class="pre">Structure</span> <span class="pre">Ranking</span></code> (<code class="docutils literal notranslate"><span class="pre">atom3d_psr</span></code>)</a></li>
<li><a class="reference internal" href="#deep-sea-protein-classification-deep-sea-proteins"><code class="xref py py-class docutils literal notranslate"><span class="pre">Deep</span> <span class="pre">Sea</span> <span class="pre">Protein</span> <span class="pre">Classification</span></code> (<code class="docutils literal notranslate"><span class="pre">deep_sea_proteins</span></code>)</a></li>
<li><a class="reference internal" href="#enzyme-commission-number-prediction-ec-reaction"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enzyme</span> <span class="pre">Commission</span> <span class="pre">Number</span> <span class="pre">Prediction</span></code> (<code class="docutils literal notranslate"><span class="pre">ec_reaction</span></code>)</a></li>
<li><a class="reference internal" href="#fold-classification-fold-family-fold-superfamily-fold-fold"><code class="xref py py-class docutils literal notranslate"><span class="pre">Fold</span> <span class="pre">Classification</span></code> (<code class="docutils literal notranslate"><span class="pre">fold-family</span></code>, <code class="docutils literal notranslate"><span class="pre">fold-superfamily</span></code>, <code class="docutils literal notranslate"><span class="pre">fold-fold</span></code>)</a></li>
<li><a class="reference internal" href="#gene-ontology-go-bp-go-cc-go-mf">Gene Ontology (<code class="docutils literal notranslate"><span class="pre">go-bp</span></code>, <code class="docutils literal notranslate"><span class="pre">go-cc</span></code>, <code class="docutils literal notranslate"><span class="pre">go-mf</span></code>)</a></li>
</ul>
</li>
<li><a class="reference internal" href="#node-level-datasets">Node-level Datasets</a><ul>
<li><a class="reference internal" href="#atom3d-residue-identity-prediction-atom3d-res">Atom3D Residue Identity Prediction (<code class="docutils literal notranslate"><span class="pre">atom3d_res</span></code>)</a></li>
<li><a class="reference internal" href="#ccpdb-ligand-binding-ccpdb-ligand">CCPDB Ligand Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_ligand</span></code>)</a></li>
<li><a class="reference internal" href="#ccpdb-metal-binding-ccpdb-metal">CCPDB Metal Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_metal</span></code>)</a></li>
<li><a class="reference internal" href="#ccpdb-nucleic-acid-binding-ccpdb-nucleic">CCPDB Nucleic Acid Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_nucleic</span></code>)</a></li>
<li><a class="reference internal" href="#ccpdb-nucleotide-binding-ccpdb-nucleotides">CCPDB Nucleotide Binding (<code class="docutils literal notranslate"><span class="pre">ccpdb_nucleotides</span></code>)</a></li>
<li><a class="reference internal" href="#post-translational-modifications-ptm">Post Translational Modifications (<code class="docutils literal notranslate"><span class="pre">ptm</span></code>)</a></li>
<li><a class="reference internal" href="#ppi-site-prediction-masif-site">PPI Site Prediction (<code class="docutils literal notranslate"><span class="pre">masif_site</span></code>)</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../_static/jquery.js?v=5d32c60e"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
    <script src="../_static/documentation_options.js?v=938c9ccc"></script>
    <script src="../_static/doctools.js?v=888ff710"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/scripts/furo.js?v=32e29ea5"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/tabs.js?v=3ee01567"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>